src/lib/numbers.ts

```diff
@@ -495,3 +496,34 @@ export function numberToBigint(value: number, decimals: number) {
 
   return negative ? -res : res;
 }
+
+export function calculatePriceDecimals(price?: bigint, decimals = USD_DECIMALS) {
+  if (price === undefined || price === 0n) return 2;
+  const priceNumber = Number(price.toString()) / Math.pow(10, decimals);
+
+  if (isNaN(priceNumber)) return 2;
+  if (priceNumber >= 1000) return 2;
+  if (priceNumber >= 100) return 3;
+  if (priceNumber >= 1) return 4;
+  if (priceNumber >= 0.1) return 5;
+  if (priceNumber >= 0.01) return 6;
+  if (priceNumber >= 0.0001) return 7;
+
+  return 8;
+}
+
+export function formatUsdPrice(price?: bigint, opts: Parameters<typeof formatUsd>[1] = {}) {
+  if (price === undefined || price < 0n) {
+    if (isDevelopment()) {
```

Comment:
I think we tollerate undefined though, at least our code organized that way right now
so I'd do `price < 0 && isDevelopment()`